### PR TITLE
swarm/pss: reorganization of package structure

### PR DESCRIPTION
**Pull request by nolash**
Due to cyclic dependency issues PssClient is now separated from the Pss core functionality package.

### CODE REVIEW

**Comment by lmars** [comment]
Please drop all changes to `node/` and `p2p/*` packages, there is no longer a need for them to reference pss directly since the services will now do this directly in both their initializer and start method (e.g. like [this](https://github.com/ethersphere/go-ethereum/blob/network-testing-framework-internode-rpc/swarm/pss/simulations/service_test.go#L139-L147) and [this](https://github.com/ethersphere/go-ethereum/blob/network-testing-framework-internode-rpc/swarm/pss/simulations/service_test.go#L174-L180)).

This then raises the question whether this re-organization is even necessary?

File: swarm/pss/client/client_test.go [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+		Created: time.Now(),
+	}
+	
+	topic := pss.NewTopic(pss.PssPingProtocol.Name, int(pss.PssPingProtocol.Version))
+	client.AddPssPeer(potaddr, pss.PssPingProtocol)
+	nid, _ := discover.HexID("0x00")
+	p := p2p.NewPeer(nid, fmt.Sprintf("%v", potaddr), []p2p.Cap{})
+	pp := protocols.NewPeer(p, client.peerPool[topic][potaddr], pss.PssPingProtocol)
+	pp.Send(msg)
+	<-client.ctx.Done()
```
Comment (zelig) [inline swarm/pss/client/client_test.go:116]: same as in incoming

**Review by zelig — CHANGES_REQUESTED** [review]

File: swarm/pss/api/pssapi.go [hunk 2]
```diff
@@ -61,8 +55,8 @@ func (pssapi *PssAPI) NewMsg(ctx context.Context, topic PssTopic) (*rpc.Subscrip
 }
 
 // SendRaw sends the message (serialized into byte slice) to a peer with topic
-func (pssapi *PssAPI) SendRaw(topic PssTopic, msg PssAPIMsg) error {
-	err := pssapi.Pss.Send(msg.Addr, topic, msg.Msg)
+func (pssapi *PssAPI) SendRaw(topic pss.PssTopic, msg pss.PssAPIMsg) error {
```
[the remark this answers is not in the document]
Comment (nolash) [reply swarm/pss/api/pssapi.go:58]: actually no, cos it mysteriously collides with PssAPI.Pss.Send if we do that. I suggest SendPss

File: swarm/pss/api/pssapi.go [hunk 3]
```diff
@@ -7,35 +7,29 @@ import (
...
-type PssAPIMsg struct {
-	Msg  []byte
-	Addr []byte
+func NewPssAPI(ps pss.PssAdapter) *PssAPI {
+	return &PssAPI{PssAdapter: ps}
 }
 
 // NewMsg API endpoint creates an RPC subscription
-func (pssapi *PssAPI) NewMsg(ctx context.Context, topic PssTopic) (*rpc.Subscription, error) {
+func (pssapi *PssAPI) NewMsg(ctx context.Context, topic pss.PssTopic) (*rpc.Subscription, error) {
```
[the remark this answers is not in the document]
Comment (nolash) [reply swarm/pss/api/pssapi.go:24]: If SendPss (see below) then ReceivePss instead?

**Comment by lmars** [comment]
Given all of the types are in the `pss` package, I think we should drop the `Pss` prefix from everything, so it would be `pss.API`, `pss.Msg` and `pss.Topic` rather than `pss.PssAPI`, `pss.PssMsg` and `pss.PssTopic`.

**Pull request closed without merging**